File: examples/runtimes/java/src/main/java/com/blok/nodes/HelloWorldNode.java
package com.blok.nodes;

import com.blok.runtime.Blok;

import java.time.Instant;
import java.util.HashMap;
import java.util.Map;

/**
 * HelloWorldNode is an example Blok node in Java
 */
public class HelloWorldNode implements Blok.NodeHandler {

    @Override
    public Object execute(Blok.Context context, Map<String, Object> config) throws Exception {
        // Get name from request body or use default
        String name = "World";

        if (context.request.body != null && context.request.body instanceof Map) {
            @SuppressWarnings("unchecked")
            Map<String, Object> bodyMap = (Map<String, Object>) context.request.body;
            if (bodyMap.containsKey("name")) {
                name = (String) bodyMap.get("name");
            }
        }

        // Get greeting prefix from config or use default
        String prefix = "Hello";
        if (config != null && config.containsKey("prefix")) {
            prefix = (String) config.get("prefix");
        }

        String message = prefix + ", " + name + "!";

        // Store in context vars for downstream nodes
        context.vars.put("greeting", message);
        context.vars.put("timestamp", Instant.now().getEpochSecond());

        // Return response
        Map<String, Object> response = new HashMap<>();
        response.put("message", message);
        response.put("timestamp", Instant.now().toString());
        response.put("language", "Java");

        return response;
    }
}

File: sdks/csharp/src/Nanoservice.Core/Nodes/HelloWorldNode.cs
using System.Text.Json;
using Nanoservice.Core.Node;

namespace Nanoservice.Core.Nodes;

/// <summary>
/// HelloWorldNode greets the user with a configurable prefix.
/// </summary>
public class HelloWorldNode : INodeHandler
{
    public Task<JsonElement> ExecuteAsync(Types.Context ctx, Dictionary<string, JsonElement> config)
    {
        // Get name from body, default to "World"
        var name = "World";
        if (ctx.Request.Body.ValueKind == JsonValueKind.Object &&
            ctx.Request.Body.TryGetProperty("name", out var nameProp) &&
            nameProp.ValueKind == JsonValueKind.String)
        {
            name = nameProp.GetString() ?? "World";
        }

        // Get prefix from config, default to "Hello"
        var prefix = "Hello";
        if (config.TryGetValue("prefix", out var prefixProp) &&
            prefixProp.ValueKind == JsonValueKind.String)
        {
            prefix = prefixProp.GetString() ?? "Hello";
        }

        var message = $"{prefix}, {name}!";

        // Store greeting in context vars
        ctx.SetVar("greeting", message);

        var result = JsonSerializer.Serialize(new
        {
            message,
            timestamp = DateTime.UtcNow.ToString("O"),
            language = "csharp"
        });

        var element = JsonDocument.Parse(result).RootElement.Clone();
        return Task.FromResult(element);
    }
}
